newIDE/app/src/ProjectManager/ProjectManagerItems.js

```diff
@@ -0,0 +1,304 @@
+    {muiTheme => {
+      const {
+        error,
+        leftIcon,
+        onRefresh,
+        indentNestedItems,
+        ...otherProps
+      } = props;
+      return (
+        <ListItem
+          style={{
+            backgroundColor: muiTheme.listItem.groupBackgroundColor,
+            borderBottom: `1px solid ${muiTheme.listItem.separatorColor}`,
+          }}
+          nestedListStyle={
+            indentNestedItems ? undefined : styles.noIndentNestedList
+          }
+          {...otherProps}
+          leftIcon={error ? <WarningIcon /> : leftIcon}
+          displayReloadButton={!!error}
+          onReload={onRefresh}
+          reloadButtonTooltip={`An error has occured in functions. Click to reload them.`}
+        />
+      );
+    }}
+  </ThemeConsumer>
+);
+
+type ItemProps = {|
+  primaryText: string,
+  editingName: boolean,
+  leftIcon?: ?React.Node,
+  onEdit: () => void,
+  onDelete: () => void,
+  addLabel: string,
+  onAdd: () => void,
+  onRename: string => void,
+  onEditName: () => void,
+  onCopy: () => void,
+  onCut: () => void,
+  onPaste: () => void,
+  onDuplicate: () => void,
+  canPaste: () => boolean,
+  canMoveUp: boolean,
+  onMoveUp: () => void,
+  canMoveDown: boolean,
+  onMoveDown: () => void,
+  buildExtraMenuTemplate?: (i18n: I18nType) => Array<MenuItemTemplate>,
+  style?: ?Object,
+|};
+
+export const Item = ({
+  primaryText,
+  editingName,
+  leftIcon,
+  onEdit,
+  onDelete,
+  addLabel,
+  onAdd,
+  onRename,
+  onEditName,
+  onCopy,
+  onCut,
+  onPaste,
+  onDuplicate,
+  canPaste,
+  canMoveUp,
+  onMoveUp,
+  canMoveDown,
+  onMoveDown,
+  buildExtraMenuTemplate,
+  style,
+}: ItemProps) => {
+  const textField = React.useRef<?TextField>(null);
+
+  React.useEffect(
+    () => {
+      if (editingName)
+        setTimeout(() => {
+          if (textField.current) textField.current.focus();
+        }, 100);
+    },
+    [editingName]
+  );
+
+  const label = editingName ? (
+    <TextField
+      id="rename-item-field"
+      margin="none"
+      ref={textField}
+      defaultValue={primaryText}
+      onBlur={e => onRename(e.currentTarget.value)}
+      onKeyPress={event => {
+        if (shouldValidate(event)) {
+          if (textField.current) textField.current.blur();
+        }
+      }}
+      fullWidth
+      style={styles.itemTextField}
+    />
+  ) : (
+    <div style={textEllispsisStyle} title={primaryText}>
+      {primaryText}
+    </div>
+  );
+
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
+        <I18n>
+          {({ i18n }) => (
+            <ListItem
+              style={{
+                borderBottom: `1px solid ${muiTheme.listItem.separatorColor}`,
+                ...style,
+              }}
+              primaryText={label}
+              leftIcon={leftIcon}
+              displayMenuButton
+              buildMenuTemplate={(i18n: I18nType) => [
+                {
+                  label: i18n._(t`Edit`),
+                  click: () => onEdit(),
```

Comment:
this can be just `onEdit` (and all the others below)

Reply:
True. I copy-pasted the existing, I don't know why it was like that at first.